src/Umbraco.Infrastructure/Migrations/Install/DatabaseBuilder.cs

```diff
@@ -102,20 +102,25 @@ public bool CanConnect(string? connectionString, string providerName)
 
         public bool HasSomeNonDefaultUser()
         {
+            if (_scopeAccessor.AmbientScope is null)
+            {
+                throw new InvalidOperationException("Cannot execute without a valid AmbientScope.");
+            }
```

Comment:
I moved this  check into the using block, since the ambientscope is actually null  here, it's just a weird way for us to "cast" to the infrastructure scope 😓